infer/inference.cc

```diff
@@ -18,17 +18,10 @@ bool Inference::willRun(core::Context ctx, core::LocOffsets loc, core::MethodRef
     }
 
     const auto &methodData = method.data(ctx);
-    auto name = methodData->name;
-    auto isMangleRenameOverload = name.kind() == core::NameKind::UNIQUE &&
-                                  name.dataUnique(ctx)->uniqueNameKind == core::UniqueNameKind::MangleRenameOverload;
-    if (isMangleRenameOverload || methodData->flags.isOverloaded) {
```

Comment:
I have thoroughly no clue how this code works, but I'm happy to just delete it move on.

Reply:
We didn't previously mark the `MangleRenameOverload` as overloaded, so we had to rely on the name to know if we had found the definition instead of the first sig. The code a few lines down explicitly looks up the first sig when we discover the `def` was found instead.